spark/CMakeLists.txt

```diff
@@ -28,82 +28,98 @@ project(spark-integration)
-
-  if (NOT SPARK_HPCC)
-    message(FATAL_ERROR "Missing Spark HPCC Jar file. Set it with -DSPARK_HPCC=<value>")
-  endif (NOT SPARK_HPCC)
-
-
-  message("")
-  message("SPARK VERSION: ${SPARK_VERSION}")
-  message("HADOOP VERSION: ${HADOOP_VERSION}")
-  message("SPARK Package URI: ${SPARK_URL}")
-  message("")
-
-  if(SHA512)
-     set( SHA512STRING "${SHA512}" )
-  else(SHA512)
-     set( SHA512STRING "DC3A97F3D99791D363E4F70A622B84D6E313BD852F6FDBC777D31EAB44CBC112CEEAA20F7BF835492FB654F48AE57E9969F93D3B0E6EC92076D1C5E1B40B4696" )
-  endif(SHA512)
-
-  string(TOLOWER "${SHA512STRING}" SHA512STRING)
-  set(URL_HASHSTRING "SHA512=${SHA512STRING}")
-
-  message("${URL_HASHSTRING}")
-
-  include(ExternalProject)
-  ExternalProject_Add(
-    fetch-spark-source
-    URL ${SPARK_URL}
-    URL_HASH ${URL_HASHSTRING}
-    TIMEOUT 300
-    DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/downloads
-    SOURCE_DIR ${CMAKE_BINARY_DIR}/downloads/spark-hadoop
-    CONFIGURE_COMMAND ""
-    BUILD_COMMAND ""
-    INSTALL_COMMAND "")
-
-  add_custom_target(
-    spark-source ALL
-    DEPENDS fetch-spark-source
-    COMMAND ${CMAKE_COMMAND} -E remove -f ${SPARK_PACKAGE_NAME}.tgz
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/downloads)
-
-  install(
-    DIRECTORY "${CMAKE_BINARY_DIR}/downloads/spark-hadoop"
-    COMPONENT runtime
-    DESTINATION "externals"
-  )
-
-  install(
-    FILES "${CMAKE_BINARY_DIR}/${SPARK_HPCC}"
-    COMPONENT runtime
-    DESTINATION "jars/spark/"
-  )
+    if(SPARK_URL)
+        string( REPLACE "\/" ";" SPARK_URL_LIST ${SPARK_URL} )
+        list( GET SPARK_URL_LIST "-1"  spark_file_name_w_ext )
+        string( REPLACE "\.tgz" "" spark_file_name_only ${spark_file_name_w_ext} )
+        string( REPLACE "-" ";" spark_file_name_parts ${spark_file_name_only} )
+        list( GET spark_file_name_parts 1 SPARK_VERSION )
+        list( GET spark_file_name_parts 3 hadoop_version_w_head )
+        string( REPLACE "hadoop" "" HADOOP_VERSION ${hadoop_version_w_head} )
+
+    else(SPARK_URL)
+        if (NOT HADOOP_VERSION)
+            set(HADOOP_VERSION 2.7)
+        endif (NOT HADOOP_VERSION)
+
+        if (NOT SPARK_VERSION)
+            set(SPARK_VERSION 2.3.1)
+        endif (NOT SPARK_VERSION)
+
+        if (NOT SPARK_URL_BASE)
+            set(SPARK_URL_BASE  "http://apache.mirrors.pair.com")
+        endif (NOT SPARK_URL_BASE)
+
+        set(SPARK_PACKAGE_NAME "spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}")
+        set(SPARK_URL  "${SPARK_URL_BASE}/spark/spark-${SPARK_VERSION}/${SPARK_PACKAGE_NAME}.tgz")
+    endif(SPARK_URL)
+
+    set(MAVEN_REPO "https://repo1.maven.org/maven2")
+    set(SONATYPE_SNAPSHOT_REPO "https://oss.sonatype.org/content/repositories/snapshots")
+    set(HPCC_NAMESPACE "org/hpccsystems")
+    set(SPARK_HPCC_VERSION "7.0.0-beta-1")
+    set(SPARK_HPCC_JAR "spark-hpcc-${SPARK_HPCC_VERSION}.jar")
+    set(WSCLIENT_VERSION "7.0.0-SNAPSHOT")
+    set(WSCLIENT_JAR "wsclient-7.0.0-20180815.193749-3-jar-with-dependencies.jar")
```

Comment:
how do we update this file when the latest version comes out?
Is there a non-timestamped version of the file so we can make this setting more dynamic?

Reply:
No there's not a non-timestamped version.  But what we should do is try to get out an official wsclient sooner rather than later, and pull from maven central rather than the sonatype snapshot repositories.

Reply:
absolutely. but we do need to provide a mechanism to target snapshots as well (will be required during development cycles)

Reply:
Perhaps we need to provide a mechanism to pick up local files.